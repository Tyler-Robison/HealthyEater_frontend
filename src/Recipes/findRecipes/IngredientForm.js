import React from "react";
import { useFormik } from "formik";
import ingredientValidate from './ingredientValidate'

import './IngredientForm.css'


const IngredientForm = ({ ingredientsList, setIngredientsList }) => {
    const validate = ingredientValidate

    const formik = useFormik({
        initialValues: {
            ingredient: ''
        },
        validate,
        onSubmit: values => addIngredient(values),
    })

    const addIngredient = (values) => {
        console.log('adding ing***************')
        const { ingredient } = values;
        if (ingredient === '') return
        console.log('ingredient', ingredient)
        formik.resetForm();
        setIngredientsList([...ingredientsList, ingredient]);
    }


    return (
        <form className="AddIngredient-Form" onSubmit={formik.handleSubmit}>

            <input
                role=''
                className="test-input"
                id="ingredient"
                name="ingredient"
                type='text'
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
                value={formik.values.ingredient}
                placeholder='Enter Ingredient'
            />

            {formik.touched.ingredient && formik.errors.ingredient && (
                <div>{formik.errors.ingredient}</div>
            )}

            <button className="general-btn" type="submit">Add Ingredient</button>
        </form>
    )
}

export default IngredientForm;