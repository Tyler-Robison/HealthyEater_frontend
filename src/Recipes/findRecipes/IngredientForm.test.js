import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import IngredientForm from './IngredientForm';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'

// 
const DummyComponent = (listState = []) => {
    const [ingredientsList, setIngredientsList] = useState(listState)

    return (
        <IngredientForm ingredientsList={ingredientsList}
            setIngredientsList={setIngredientsList} />
    )
}

it("renders without crashing", function () {
    <MemoryRouter>
        <DummyComponent />
    </MemoryRouter>
});


it("matches snapshot with blank ingredientList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

// will have identical HTML structure 
// making sure it still renders correctly with list items. 
it("matches snapshot with filled ingredientList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent listState={['cheese', 'eggs']} />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});