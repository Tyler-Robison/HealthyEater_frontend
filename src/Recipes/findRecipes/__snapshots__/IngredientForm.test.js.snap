// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with blank ingredientList 1`] = `
<DocumentFragment>
  <form
    class="AddIngredient-Form"
  >
    <input
      class="test-input"
      id="ingredient"
      name="ingredient"
      placeholder="Enter Ingredient"
      role=""
      type="text"
      value=""
    />
    <button
      class="general-btn"
      type="submit"
    >
      Add Ingredient
    </button>
  </form>
</DocumentFragment>
`;

exports[`matches snapshot with filled ingredientList 1`] = `
<DocumentFragment>
  <form
    class="AddIngredient-Form"
  >
    <input
      class="test-input"
      id="ingredient"
      name="ingredient"
      placeholder="Enter Ingredient"
      role=""
      type="text"
      value=""
    />
    <button
      class="general-btn"
      type="submit"
    >
      Add Ingredient
    </button>
  </form>
</DocumentFragment>
`;
